### PR TITLE
Finalize lm examples

### PR DESCRIPTION
**Pull request by sgugger**
# What does this PR do?

This PR adds a `run_mlm_wwm` script for an example of MLM with whole word masking, which was the last kind of examples supporter by `run_language_modeling`.

As a result, it moves this script to `contrib/legacy/` and udpates the README to document how to use all the new example scripts.

I also reworked a tiny bit the table of tasks in the main README to include whether or not the examples leverage the Datasets library.

### CODE REVIEW

File: examples/language-modeling/run_mlm_wwm.py [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+    """
+
+    train_file: Optional[str] = field(default=None, metadata={"help": "The input training data file (a text file)."})
+    validation_file: Optional[str] = field(
+        default=None,
+        metadata={"help": "An optional input evaluation data file to evaluate the perplexity on (a text file)."},
+    )
+    train_ref_file: Optional[str] = field(
+        default=None,
+        metadata={"help": "An optional input train ref data file for whole word mask in Chinese."},
```
Comment (LysandreJik) [inline examples/language-modeling/run_mlm_wwm.py:100]:
```suggestion
        metadata={"help": "An optional input train ref data file for whole word masking in Chinese."},
```

File: examples/language-modeling/run_mlm_wwm.py [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+    )
+    validation_ref_file: Optional[str] = field(
+        default=None,
+        metadata={"help": "An optional input validation ref data file for whole word mask in Chinese."},
```
Comment (LysandreJik) [inline examples/language-modeling/run_mlm_wwm.py:104]:
```suggestion
        metadata={"help": "An optional input validation ref data file for whole word masking in Chinese."},
```

**Review by LysandreJik — APPROVED** [review]
Cool, this is very clean! Love the new README too.

**Review by thomwolf — APPROVED** [review]
Very nice!

**Pull request merged**